python/src/scipp/table.py

```diff
@@ -31,175 +21,133 @@ def value_to_string(val, precision=3):
     return text
 
 
-def append_with_text(parent, name, text, attrib=style_border_right):
-    el = et.SubElement(parent, name, attrib=attrib)
-    el.text = text
-
-
-def table_ds(dataset):
-    coords = dataset.coords
-    ndims = len(coords)
-    if ndims > 1:
-        raise RuntimeError("Only 0D & 1D datasets can be rendered as a table")
-
-    body = et.Element('body')
-    headline = et.SubElement(body, 'h3')
-    if isinstance(dataset, sp.Dataset):
-        headline.text = 'Dataset:'
-    else:
-        headline.text = 'DataProxy:'
-
-    datum1d = defaultdict(list)
-    datum0d = defaultdict(list)
-    coords1d = None
-    for name, var in dataset:
-        if len(var.coords) == 1:
-            datum1d[name] = var
-            coords1d = var.coords[var.dims[0]]
-
+def table_from_dataset(dataset, coord_dim=None, is_hist=False):
+    # border style
+    bstyle = "style='border: 1px solid black;"
+    cstyle = [bstyle + "background-color: #ADF3E0;'",
```

Comment:
Good additions for highlighting. Though I'm not 100% on colour palette. A future addition might be to allow the caller to customise certain aspects of the style - not for this PR though.